ngx_http_sxg_filter_module.c

```diff
@@ -537,15 +556,20 @@ static ngx_int_t ngx_http_sxg_header_filter(ngx_http_request_t* req) {
     if (ctx == NULL) {
```

Comment:
This check of `ctx` confuses me due to the above `--ctx->subresources`.

Reply:
This is the entry point of every request from internet.
So that `ctx` is usually `NULL` because it has not been allocated yet. This check traps allocation failure from memory pool.

`--ctx->subresources` must be called after the ctx is allocated. It may be problem of readability of order. Some refactoring would be required, but not a bug.